### DISCUSSION TITLE
Symfony and React within a Docker-Container: Could not find the entrypoints file from Webpack

### DISCUSSION THREAD

**Discussion by chris576** [symfony/symfony#50817] [Q&A]
I am trying to run nginx, symfony and react within a docker container. 

I am getting this error when loading the page: 

    An exception has been thrown during the rendering of a template ("Could not find the entrypoints file from Webpack: the file "/var/www/html/public/build/entrypoints.json" does not exist.").

I have consulted the following documentaries yet: 

https://symfony.com/doc/current/frontend/encore/faq.html#uncaught-referenceerror-webpackjsonp-is-not-defined

https://symfony.com/doc/current/frontend/encore/simple-example.html#configuring-encore-webpack

My package.json: 

    {
      "dependencies": {
        "@babel/plugin-proposal-class-properties": "^7.18.6",
        "@babel/plugin-transform-runtime": "^7.22.5",
        "prop-types": "^15.8.1",
        "react": "^18.2.0",
        "react-dom": "^18.2.0",
        "react-router-dom": "^6.14.0"
      },
      "devDependencies": {
        "@babel/plugin-syntax-jsx": "^7.22.5",
        "@babel/preset-react": "^7.22.5",
        "@symfony/webpack-encore": "^4.3.0",
        "axios": "^1.4.0"
      },
      "scripts": {
        "build": "webpack"
      }
    }

Dockerfile:

    # Basisimage
    FROM php:8.1-fpm
    
    # Installiere Abhängigkeiten
    RUN apt-get update && apt-get install -y \
        libicu-dev \
        libpq-dev \
        && docker-php-ext-install \
        pdo_mysql \
        pdo_pgsql \
        intl
    
    RUN curl -sL https://deb.nodesource.com/setup_14.x | bash -
    
    RUN apt-get install -y nodejs npm
    RUN npm install -g yarn
    
    WORKDIR /var/www/html
    
    COPY . .
    
    # Installiere Composer
    RUN curl -sS https://getcomposer.org/installer | php -- --install-dir=/usr/local/bin --filename=composer
    
    # Installiere die Composer-Abhängigkeiten
    RUN composer install --no-scripts --no-autoloader
    
    # Setze die Dateiberechtigungen
    RUN chown -R www-data:www-data /var/www/html/var
    
    RUN yarn add webpack-notifier@^1.15.0 --dev
    
    RUN yarn build
    
    # Exponiere den Container-Port
    EXPOSE 9000
    
    # Starte den PHP-FPM-Server
    CMD ["php-fpm"]

The webpack.config.js: 

    var Encore = require('@symfony/webpack-encore');
    
    // It's useful when you use tools that rely on webpack.config.js file.
    if (!Encore.isRuntimeEnvironmentConfigured()) {
        Encore.configureRuntimeEnvironment(process.env.NODE_ENV || 'dev');
    }
    
    Encore
        // directory where compiled assets will be stored
        .setOutputPath('public/build/')
        // public path used by the web server to access the output path
        .setPublicPath('/build')
        .enableReactPreset()
        // only needed for CDN's or sub-directory deploy
        //.setManifestKeyPrefix('build/')
    
        /*
         * ENTRY CONFIG
         *
         * Add 1 entry for each "page" of your app
         * (including one that's included on every page - e.g. "app")
         *
         * Each entry will result in one JavaScript file (e.g. app.js)
         * and one CSS file (e.g. app.css) if you JavaScript imports CSS.
         */
        .addEntry('app', './assets/app.js')
    
        // When enabled, Webpack "splits" your files into smaller pieces for greater optimization.
        .splitEntryChunks()
    
        // will require an extra script tag for runtime.js
        // but, you probably want this, unless you're building a single-page app
        .enableSingleRuntimeChunk()
    
        /*
         * FEATURE CONFIG
         *
         * Enable & configure other features below. For a full
         * list of features, see:
         * https://symfony.com/doc/current/frontend.html#adding-more-features
         */
        .cleanupOutputBeforeBuild()
        .enableBuildNotifications()
        .enableSourceMaps(!Encore.isProduction())
        // enables hashed filenames (e.g. app.abc123.css)
        .enableVersioning(Encore.isProduction())
    
        // enables @babel/preset-env polyfills
        .configureBabel(() => {}, {
            useBuiltIns: 'usage',
            corejs: 3
        })
    ;
    
    module.exports = Encore.getWebpackConfig();

If anything was working correctly, I Hello World should be printed in a h1 react-tag. 

Due to the documentary, I have consulted, the build process must be correct. Have looked within the docker container, if the file was build somewhere else, but it did not generate at all. 

The whole project can be found here: https://gitlab.technik-emden.de/ch6447/HardwareBookingSystem

**Comment by ghost** [comment]


**Reply by chris576** [reply]
It is dev env. What do you mean by second question?

**Reply by ghost** [reply]
Ok, i took a look on your gitlab repo. In compose file you bind local project root to /var/www/html inside container - this way you override what docketfile has built in this location. Now you have to run yarn build again when container is running to dump files on your localhost.

**Reply by ghost** [reply]
Btw. This repo is incomplete. Where is the assets folder or webpack.config.js file? Please keep in sync the code you past here and in the repo you provided.

**Reply by chris576** [reply]
Have committed now the other files and changes the composer. The build process is not working, while I still cant see "Hello World" on the landing page.

**Reply by chris576** [reply]
Thank you a lot for your help, @javaDeveloperKid. I am new to docker, so I could have spent weeks troubleshooting this.

**Reply by ghost** [reply]
Dude, why did you remove volume from docker-compose?
```yaml
    volumes:
      - .:/var/www/html
```

I'm sorry but I don't have time to teach you docker basics. Good luck.

**Reply by chris576** [reply]
@javaDeveloperKid I have removed it, because you said, that it would overwrite my build. So I want to see, what difference it makes, if I removed it.
